crates/typst/src/diag.rs

```diff
@@ -228,6 +239,49 @@ impl From<SyntaxError> for SourceDiagnostic {
     }
 }
 
+/// The identifier of a [`SourceDiagnostic`].
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub enum Identifier {
```

Comment:
The name of this enum is maybe a bit too generic.

Reply:
The thought I had is that it's already scoped under `diag::`, so `diag::Identifier` does it justice. We could, however, prefix it with `Diagnostic` or similar.

Reply:
Resolving due to https://github.com/typst/typst/pull/4459#discussion_r1688231225